pkgs/development/interpreters/python/hooks/pytest-check-hook.sh

```diff
@@ -50,25 +50,25 @@ function pytestCheckPhase() {
       args+=" --ignore=\"$path\""
     done
     args+=" ${pytestFlagsArray[@]}"
-    eval "@pythonCheckInterpreter@ $args"
+
+    PYTHONPATH_TMP="@prefixPythonPath@:$PYTHONPATH:@pytest@"
+
+    PYTHONPATH=$PYTHONPATH_TMP eval "@pythonCheckInterpreter@ $args"
 
     runHook postCheck
     echo "Finished executing pytestCheckPhase"
 }
 
-if [ -z "${dontUsePytestCheck-}" ] && [ -z "${installCheckPhase-}" ]; then
-    echo "Using pytestCheckPhase"
-    preDistPhases+=" pytestCheckPhase"
+# because pytestCheckHook has replaced setuptoolsCheckHook,
+# the hook must still be executed even if dontUsePytestCheck is set,
+# as long as dontUseSetuptoolsCheck is not set
```

Comment:
if we do actually remove the hook, then I think we should stop listening to `dontUseSetuptoolsCheck` as well.